### PR TITLE
Update Docker deployment scripts

### PR DESCRIPTION
**Pull request by danlamanna**
This is just a new Dockerfile, soon to come is the infrastructure for deploying this with SMQTK.

### CODE REVIEW

**Comment by danlamanna** [comment]
@jeffbaumes  to test:

```
docker-compose up
```

From the directory that `docker-compose.yml` is in.

Then visit localhost at whatever port `imagespace-girder` got mapped to (login is girder/girder).

File: scripts/deploy/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  imagespace-solr:
+    image: solr:6.0
+    networks:
+      - imagespace-network
+    entrypoint:
+      - docker-entrypoint.sh
+      - solr-precreate
+      - imagespace
+  imagespace-girder:
+    image: kitware/image_space:latest
```
Comment (chrismattmann) [inline scripts/deploy/docker-compose.yml]: why does this depend on something specific to kitware? Shouldn't it be memex-explorer/image_space:latest?

File: scripts/deploy/imagespace/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# Docker image: kitware/imagespace
```
Comment (chrismattmann) [inline scripts/deploy/imagespace/Dockerfile]: should be memex-explorer/imagespace

File: scripts/deploy/imagespace/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+# This spins up an install of ImageSpace based on the Girder Docker image.
+
+# Requirements:
+# - Directory of images to mount for serving (at IMAGE_SPACE_IMAGE_DIR)
+# - Solr instance with URL to core as IMAGE_SPACE_SOLR
```
Comment (chrismattmann) [inline scripts/deploy/imagespace/Dockerfile:21]: please add ALv2 license header.

File: scripts/deploy/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,26 @@
+version: "2"
```
Comment (chrismattmann) [inline scripts/deploy/docker-compose.yml:15]: please add ALv2 license header.

File: scripts/deploy/imagespace/imagespace-entrypoint.sh [hunk 5]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
```
Comment (chrismattmann) [inline scripts/deploy/imagespace/imagespace-entrypoint.sh:1]: please add ALv2 license header.

File: scripts/deploy/imagespace/bootstrap-imagespace.py [hunk 6]
```diff
@@ -0,0 +1,44 @@
+import json
```
Comment (chrismattmann) [inline scripts/deploy/imagespace/bootstrap-imagespace.py:15]: please add ALv2 license header.

**Comment by chrismattmann** [comment]
@danlamanna some simple cleanup and I think this is ready to go I'll try and test this week.

**Comment by jeffbaumes** [comment]
@danlamanna I was able to access it when I exposed the port to the host, see my latest commit (i randomly chose port 8989 since 8080 is common).

**Comment by chrismattmann** [comment]
+1 @jeffbaumes 👍 on the port

**Comment by danlamanna** [comment]
Fixed up some of these, but I'm setting this to WIP for the moment - needs some more work.

**Comment by jeffbaumes** [comment]
@danlamanna check that the fixes for Mac don't break Linux. Mac's Unix tools are messed up.

**Comment by danlamanna** [comment]
@jeffbaumes It did break Linux, check if my new changes work on Mac?

**Comment by jeffbaumes** [comment]
Works! LGTM

**Pull request merged**